schema-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaSchemaSerDe.java

```diff
@@ -632,6 +670,23 @@ protected Object executeRules(
     return message;
   }
 
+  private boolean skipRule(Rule rule, Headers headers) {
+    if (rule.isDisabled()) {
+      return true;
+    }
+    if (headers != null) {
+      Header header = headers.lastHeader(DlqAction.RULE_NAME);
+      if (header != null) {
+        String ruleName = new String(header.value(), StandardCharsets.UTF_8);
+        if (rule.getName().equals(ruleName)) {
+          // Don't rerun failed rule
+          return true;
+        }
+      }
```

Comment:
What does this do? Isn't `DlqAction.RULE_NAME` only present in the message that was sent to DLQ?

Reply:
This is for a deserializer which is reading from the DLQ.  In that case we don't want the rule to run again as it will fail again and we won't be able to deserialize.  Good question, I'll enhance the comment.